backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh

```diff
@@ -21,6 +21,74 @@
 #include <string>
 #include <vector>
 
+template <typename Torus>
+__global__ void smart_copy(Torus *dst, Torus *src, int32_t *id_out,
```

Comment:
I don't understand the purpose of this kernel: is it really faster than using cuda_memcpy and cuda_memset? It seems to me we could replace it with calls to these two functions.